Read sensors in the systick

Peque opened the pull request.

cua-cua commented on src/calibration.c:
```diff
@@ -107,27 +107,44 @@ void run_static_turn_right_profile(void)
+	uint32_t ticks_to_stop;
 
 	disable_walls_control();
+	side_sensors_calibration();
```
why are you using side_sensors_calibration?

Peque replied:
Removed. :+1:

cua-cua commented on src/calibration.c:
```diff
@@ -107,27 +107,44 @@ void run_static_turn_right_profile(void)
+		sleep_us(1000);
+	}
+	required_deceleration =
+	    (.3 * .3) /
```
Macro for set_target_linear_speed and these .3* .3.

Peque replied:
Created a variable to store that value (58693b9e4a6448637cbbdbccf30c05b4fb6d4dd8).

cua-cua commented on src/target/detection.c:
```diff
@@ -70,75 +68,66 @@ static void set_emitter_off(uint8_t emitter)
 
 /**
- * @brief Get sensors values with emitter on and off.
+ * @brief Update the sensors raw readings.
```
You can use adc_clear_flag from common.c file (pull request to libopencm3, it was already working for watchdog interruption flag)

Peque replied:
It seems the `adc_clear_flag()` differs from what it is written here, right? Now we have:

```c
ADC_SR(ADC1) &= ~ADC_SR_JEOC; 
```

But calling  `adc_clear_flag()` would result in:

```c
ADC_SR(ADC1) = ~ADC_SR_JEOC; 
```

Note the `&` is missing.

Is `adc_clear_flag()` broken?

Peque replied:
Changed docstring as requested: 6faa22adf5dc3d9b981773b38e1074669aaa13e0

cua-cua commented on src/target/detection.c:
```diff
@@ -70,75 +68,66 @@ static void set_emitter_off(uint8_t emitter)
+}
+
+/**
+ * @brief Update the sensors raw readings.
```
Expand the description

cua-cua commented on src/target/detection.c:
```diff
@@ -290,7 +279,7 @@ void side_sensors_calibration(void)
 		left_temp += distance[SENSOR_SIDE_LEFT_ID];
 		right_temp += distance[SENSOR_SIDE_RIGHT_ID];
-		sleep_ticks(SENSORS_SM_TICKS);
+		sleep_ticks(2);
```
why did you change it to 2? Coudl you add an explicative macro?

Peque replied:
How would you call it? It is just some small delay in between readings, it could be any other number really.

I could create a `SIDE_CALIBRATION_READINGS_DELAY_TICKS` but looks huge just for one use and it really is not important. Does it hurt to much in your eyes? :joy:

cua-cua commented on src/target/detection.c:
```diff
@@ -113,19 +225,21 @@ static void update_raw_readings(void)
+		diff = 0;
+	else
+		diff = on - off;
+	diff /= 4;
```
Why 4?

cua-cua commented on src/control.c:
```diff
@@ -9,14 +9,14 @@
  */
 static volatile float linear_acceleration = 5.;
```
why these changes on this commit?

Peque replied:
Just for now, while the robot keeps crashing into walls like a drunken robot... I will remove it before merging into master. I hope...

cua-cua commented on src/calibration.c:
```diff
@@ -143,6 +143,7 @@ void run_movement_sequence(const char *sequence)
  */
 void run_front_sensors_calibration(void)
```
good :)

Peque replied:
:smile:

Peque commented:
Moved to Portugal 2019 milestone. Added "on hold" label as this method results in less reliable readings. We probably need some hardware changes to improve that (i.e.: add capacitors next to the emitters to provide the required transient power).
